x/evidence/internal/keeper/keeper_test.go

```diff
@@ -0,0 +1,192 @@
+package keeper_test
+
+import (
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/store"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/evidence/internal/keeper"
+	"github.com/cosmos/cosmos-sdk/x/evidence/internal/types"
+	"github.com/cosmos/cosmos-sdk/x/params"
+
+	"github.com/stretchr/testify/suite"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/crypto/ed25519"
+	"github.com/tendermint/tendermint/libs/log"
+	tmtypes "github.com/tendermint/tendermint/types"
+	dbm "github.com/tendermint/tm-db"
+)
+
+type KeeperTestSuite struct {
+	suite.Suite
+
+	cms     sdk.CommitMultiStore
+	ctx     sdk.Context
+	querier sdk.Querier
+	keeper  *keeper.Keeper
+}
+
+func (suite *KeeperTestSuite) SetupTest() {
+	// create required store keys
+	keyParams := sdk.NewKVStoreKey(params.StoreKey)
```

Comment:
should we use `simapp` for integration tests as well? @colin-axner was working on those during his internship check `x/bank/internal/keeper/integration_test.go`

Reply:
This requires concrete types which I don't anticipate having in this PR.

Reply:
I'm not following, you can update the `SetupTest()` to something like:

```go
func (suite *KeeperTestSuite) SetupTest() {
	isCheckTx := false
	app := simapp.Setup(isCheckTx)

	suite.cdc = app.Codec()
	suite.ctx = app.BaseApp.NewContext(isCheckTx, abci.Header{})
	suite.keeper = &app.EvidenceKeeper
	suite.querier = keeper.NewQuerier(app.EvidenceKeeper)
}
```

You'd need to add the keeper to simapp

Reply:
Sorry, I thought you meant adding this module to the simulation. Yeah, we can do this.